Formula/clickhouse-cpp.rb

```diff
@@ -0,0 +1,78 @@
+class ClickhouseCpp < Formula
+  desc "C++ client library for ClickHouse"
+  homepage "https://github.com/ClickHouse/clickhouse-cpp#readme"
+  url "https://github.com/ClickHouse/clickhouse-cpp.git",
+    tag:      "1.5.0",
+    revision: "1415b5936a2ac2f084850b09057e05fb5798b2f1"
+  license "Apache-2.0"
+  head "https://github.com/ClickHouse/clickhouse-cpp.git",
+    branch:   "master"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"main.cpp").write <<~EOS
+      #include <clickhouse/client.h>
+
+      #include <exception>
+
+      #include <cstdio>
+      #include <cstdlib>
+
+      int main(int argc, char* argv[])
+      {
+          int exit_code = EXIT_SUCCESS;
+
+          try
+          {
+              // Expecting a typical "failed to connect" error.
+              clickhouse::Client client(
+                clickhouse::ClientOptions()
+                .SetHost("example.com")
+                .SetSendRetries(1)
+                .SetRetryTimeout(std::chrono::seconds(1))
+                .SetTcpKeepAliveCount(1)
+                .SetTcpKeepAliveInterval(std::chrono::seconds(1))
+              );
+          }
+          catch (const std::exception& ex)
+          {
+              std::fprintf(stdout, "Exception: %s\\n", ex.what());
+              exit_code = EXIT_FAILURE;
+          }
+          catch (...)
+          {
+              std::fprintf(stdout, "Exception: unknown\\n");
+              exit_code = EXIT_FAILURE;
+          }
+
+          return exit_code;
+      }
+    EOS
+
+    (testpath/"CMakeLists.txt").write <<~EOS
+      project (clickhouse-cpp-test-client LANGUAGES CXX)
+
+      set (CMAKE_CXX_STANDARD 17)
+      set (CMAKE_CXX_STANDARD_REQUIRED ON)
+
+      set (CLICKHOUSE_CPP_INCLUDE "#{include}")
+      find_library (CLICKHOUSE_CPP_LIB NAMES clickhouse-cpp-lib PATHS "#{lib}" REQUIRED NO_DEFAULT_PATH)
+
+      add_executable (test-client main.cpp)
+      target_include_directories (test-client PRIVATE ${CLICKHOUSE_CPP_INCLUDE})
+      target_link_libraries (test-client PRIVATE ${CLICKHOUSE_CPP_LIB})
+    EOS
+
+    system "cmake", "-S", testpath, "-B", (testpath/"build"), *std_cmake_args
+    system "cmake", "--build", (testpath/"build")
```

Comment:
Do we really need to use CMake to test? Can't we just compile the CPP file directly?

Reply:
This is kind of future-proof test strusture: `clickhouse-cpp` will be modified soon to generate CMake config files during the install step, so that consuming the library in client CMake projects will be more clean and easy, so it will be a one line change to test it here too.
Also, didn't want to deal with compiler directly here...